File: src/Scripting/PHP/patterns/service_locator.php
<?php declare(strict_types=1);
$check = static fn(bool $ok) => $ok ?: throw new RuntimeException('service_locator');
$services = ['clock' => static fn(): string => '12:00'];
$locate = static fn(string $name): callable => $services[$name] ?? throw new RuntimeException('missing');
$check($locate('clock')() === '12:00');

File: src/Scripting/Lua/patterns/service_locator.lua
local locator = { services = {} }
function locator:register(name, service) self.services[name] = service end
function locator:get(name) assert(self.services[name], "missing service"); return self.services[name] end
locator:register("clock", function() return 123 end)
assert(locator:get("clock")() == 123)
return true
